app/api/csv.py

```diff
@@ -3,12 +3,67 @@
 import csvx
 from io import StringIO
 from collections import OrderedDict as od
+import json
 
 
-def generate_brief_responses_csv(brief, responses):
-    def csv_cell_sanitize(text):
-        return re.sub(r"^(=|\+|-|@|!|\|{|}|\[|\]|<|,)+", '', unicode(text).strip())
+def csv_cell_sanitize(text):
+    return re.sub(r"^(;|=|\+|-|@|!|\|{|}|\[|\]|<|,)+", '', unicode(text).strip())
+
+
+def convert_to_csv(data, convertor_function, transpose=False):
+    # convert the responses to a list - each row is a dict representing a brief response
+    rows = [convertor_function(d) for d in data]
+    # add the keys of a dict as the first element in the rows list, to be used as headers
+    first = rows[0].keys() if data else []
+    first = [f.replace('_', ' ').capitalize() if '_' in f else f for f in first]
+    rows = [first] + [r.values() for r in rows]
+
+    csvdata = StringIO()
+    with csvx.Writer(csvdata) as csv_out:
+        if transpose:
+            # convert the rows to a zip'd list - this creates a structure so the csv's
+            # first column is the headers and the
+            # remaining columns contain the data for each header row
+            transposed = list(six.moves.zip_longest(*rows))
+            csv_out.write_rows(transposed)
+            return csvdata.getvalue()
+        else:
+            csv_out.write_rows(rows)
+            return csvdata.getvalue()
```

Comment:
If these two `return` statements are the same, should we have a single one outside of the `if` block?

Reply:
updated